theano/tensor/var.py

```diff
@@ -466,6 +467,25 @@ def astype(self, dtype):
 
     # SLICING/INDEXING
     def __getitem__(self, args):
+
+        def check_bool(args_el):
+            try:
+                if isinstance(args_el, (numpy.bool_, bool)) or \
+                   args_el.dtype == 'int8' or args_el.dtype == 'uint8':
+                    raise TypeError(('TensorType does not support boolean '
+                                     'mask for indexing such as tensor[x==0]. '
+                                     'If you are indexing on purpose with an '
+                                     'int8, please cast it to int32'))
```

Comment:
It would be better to also mention that you can use `non_zeros()`, in that case `tensor[(x == 0).nonzeros()]` should give the expected result.
It would also be better to mention int16 instead of int32, since it would work as well.
This change means you will need to update the doctest as well.